src/robusta/core/sinks/sink_base_params.py

```diff
@@ -55,13 +56,54 @@ def check_intervals(cls, intervals: List[ActivityInterval]):
         return intervals
 
 
-class SinkBaseParams(BaseModel):
+class RegularNotificationModeParams(BaseModel):
+    ignore_first: int = 0
+
+
+# a list of attribute names, which can be strings or dicts of the form
+# {"labels": ["app", "label-xyz"]} etc.
+GroupingAttributeSelectorListT = List[Union[str, Dict[str, List[str]]]]
+
+
+class SummaryNotificationModeParams(BaseModel):
+    threaded: bool = True
+    by: GroupingAttributeSelectorListT = ["identifier"]
+
+
+class NotificationModeParams(BaseModel):
+    regular: Optional[RegularNotificationModeParams]
+    summary: Optional[SummaryNotificationModeParams]
+    # TODO should we enforce that only one of these is set?
+
+    @root_validator
+    def validate_exactly_one_defined(cls, values: Dict):
+        if values.get("regular") and values.get("summary"):
+            raise ValueError('"regular" and "summary" notification grouping modes must not be defined at the same time')
+        if not (values.get("regular") or values.get("summary")):
+            raise ValueError('either "regular" or "summary" notification grouping mode must be defined')
+        return values
+
+
+class GroupingParams(BaseModel):
+    group_by: Optional[GroupingAttributeSelectorListT]
```

Comment:
it would be nice of `group_by` is empty, means grouped for the whole cluster, for the given interval

Reply:
Done